iree/samples/dynamic_shapes/dynamic_shapes.ipynb

```diff
@@ -0,0 +1,555 @@
+        "FH3IRpYTta2v"
+      ]
+    },
+    "kernelspec": {
+      "display_name": "Python 3",
+      "name": "python3"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "FH3IRpYTta2v"
+      },
+      "source": [
+        "##### Copyright 2021 The IREE Authors"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "mWGa71_Ct2ug",
+        "cellView": "form"
+      },
+      "source": [
+        "#@title Licensed under the Apache License v2.0 with LLVM Exceptions.\n",
+        "# See https://llvm.org/LICENSE.txt for license information.\n",
+        "# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception"
+      ],
+      "execution_count": 1,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "h5s6ncerSpc5"
+      },
+      "source": [
+        "# Dynamic Shapes\n",
+        "\n",
+        "This notebook\n",
+        "\n",
+        "1. Creates a TensorFlow program with dynamic shapes\n",
+        "2. Imports that program into IREE's compiler\n",
+        "3. Compiles the imported program to an IREE VM bytecode module\n",
+        "4. Tests running the compiled VM module using IREE's runtime\n",
+        "5. Downloads compilation artifacts for use with the native (C API) sample application"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "s2bScbYkP6VZ",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "fea0500f-d1d4-4064-feb1-5a80764d69cc"
+      },
+      "source": [
+        "#@title General setup\n",
+        "\n",
+        "import os\n",
+        "import tempfile\n",
+        "\n",
+        "ARTIFACTS_DIR = os.path.join(tempfile.gettempdir(), \"iree\", \"colab_artifacts\")\n",
+        "os.makedirs(ARTIFACTS_DIR, exist_ok=True)\n",
+        "print(f\"Using artifacts directory '{ARTIFACTS_DIR}'\")"
+      ],
+      "execution_count": 2,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Using artifacts directory '/tmp/iree/colab_artifacts'\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dBHgjTjGPOJ7"
+      },
+      "source": [
+        "## Create a program using TensorFlow and import it into IREE"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hwApbPstraWZ"
+      },
+      "source": [
+        "#@title Define a sample TensorFlow module using dynamic shapes\n",
+        "\n",
+        "import tensorflow as tf\n",
+        "\n",
+        "class DynamicShapesModule(tf.Module):\n",
+        "  # reduce_sum (dynamic input size, static output size)\n",
+        "  #   e.g. [1, 2, 3] -> 6\n",
+        "  @tf.function(input_signature=[tf.TensorSpec([None], tf.int32)])\n",
+        "  def reduce_sum(self, values):\n",
```

Comment:
same note here about dynamic sizes, and since you have all the space on the righthand side under the title it'd be nice to have at least a snippet here like you have in the readme background that if you do statically know something it's always a good thing to specify that and only use dynamic dimensions when things are really dynamic.

Reply:
copied the snippet from the readme